planner_cspace/src/planner_3d.cpp

```diff
@@ -1322,6 +1325,43 @@ class Planner3dNode
       {
         return;
       }
+      else if (is_path_switchback_)
+      {
+        static ros::Time last_time = start_.header.stamp;
+        const ros::Time time = start_.header.stamp;
+        const float dt = time.toSec() - last_time.toSec();
+        if (dt > 0.0)
+        {
+          static float last_len = std::numeric_limits<float>::quiet_NaN();
+          static float last_yaw = std::numeric_limits<float>::quiet_NaN();
+          const float len = std::hypot(
+              start_.pose.position.y - sw_pos_.pose.position.y,
+              start_.pose.position.x - sw_pos_.pose.position.x);
+          const float yaw = tf2::getYaw(start_.pose.orientation);
+          const float sw_yaw = tf2::getYaw(sw_pos_.pose.orientation);
+          float yaw_diff = yaw - sw_yaw;
+          yaw_diff = std::atan2(std::sin(yaw_diff), std::cos(yaw_diff));
+          if (len < goal_tolerance_lin_f_ && std::fabs(yaw_diff) < goal_tolerance_ang_f_)
+          {
+            // robot has arrived at the switchback point
+            is_path_switchback_ = false;
+          }
+          else if (!std::isnan(last_len) && !std::isnan(last_yaw))
+          {
+            float d_len = last_len - len;
+            float d_yaw = yaw - last_yaw;
+            d_yaw = std::atan2(std::sin(d_yaw), std::cos(d_yaw));
+            if (std::fabs(d_len) < 1e-6 && std::fabs(d_yaw) < 1e-6)
```

Comment:
How do these `1e-6` decided?

Reply:
There is no particular basis for this.
I have seen the robot (in simulation) sometimes stop a little further away from the switchback point than `goal_tolerance_lin_f`.
In this case, `waitUntil()` doesn't finish, so I decided to set `is_switchback` to false when the robot stops.
In order to determine whether or not the robot stops,  a threshold for the amount of movement was needed.
So, I tried to use `1e-6` as a small value and it worked well, so I left it as it is.

Reply:
However, since `sw_wait` has been restored, this process may no longer be necessary.

Reply:
I agree that this part can be removed. Also the check of `if (time > last_time)` may be removed.